File: Grid/Grid/Selenium_assignment_Java/src/Selenium_activity/upload_file.java
package Selenium_activity;

import org.openqa.selenium.By;
import org.openqa.selenium.JavascriptExecutor;
import org.openqa.selenium.remote.DesiredCapabilities;
import org.openqa.selenium.remote.LocalFileDetector;
import org.openqa.selenium.remote.RemoteWebDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

public class upload_file {

  public static final String AUTOMATE_USERNAME = "pratikpathare_oMjcIz";
  public static final String AUTOMATE_KEY = "EeUYtZMrb6bLzmGQoszr";
  public static final String URL = "https://" + AUTOMATE_USERNAME + ":" + AUTOMATE_KEY + "@hub-cloud.browserstack.com/wd/hub";

  public static void main(String[] args) throws Exception {

    DesiredCapabilities caps = new DesiredCapabilities();
    caps.setCapability("browser", "Chrome");
    caps.setCapability("os", "windows");
    caps.setCapability("browser_version", "95.0");
    caps.setCapability("os_version", "10");
    caps.setCapability("browserstack.sendKeys", "true");
    caps.setCapability("browserstack.debug", "true");
    caps.setCapability("name", "PHP_Java_File_Upload");

    RemoteWebDriver driver = new RemoteWebDriver(new java.net.URL(URL), caps);
    driver.setFileDetector(new LocalFileDetector());
    driver.get("http://www.fileconvoy.com/");
    driver.findElement(By.id("upfile_0")).sendKeys("/Users/pratikhemantpathare/Desktop/Demo.txt");
    driver.findElement(By.name("upload_button")).submit();
    JavascriptExecutor jse = (JavascriptExecutor)driver;
    
    driver.quit();
  }
}

File: Driver/Chrome/My_Python_code/upload_file.py
from selenium import webdriver
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.common.desired_capabilities import DesiredCapabilities
from selenium.common.exceptions import TimeoutException
from selenium.webdriver.support.ui import WebDriverWait

desired_cap = {
 'browser': 'Chrome',
 'browser_version': '95.0',
 'os': 'Windows',
 'os_version': '10',
 'browserstack.sendKeys': 'true',
 'browserstack.debug': 'true',
 'name': 'File Upload Python PHP'
}

driver = webdriver.Remote(
    command_executor='http://pratikpathare_oMjcIz:EeUYtZMrb6bLzmGQoszr@hub.browserstack.com/wd/hub',
    desired_capabilities=desired_cap)
driver.get('http://www.fileconvoy.com')
driver.find_element_by_id('upfile_0').send_keys('/Users/pratikhemantpathare/Desktop/Demo.txt')
driver.find_element_by_id('readTermsOfUse').click()
driver.find_element_by_name('upload_button').submit()

driver.quit()
